controllers/ssp_controller.go

```diff
@@ -187,8 +198,32 @@ func cleanup(request *common.Request) error {
 }
 
 func reconcileOperands(sspRequest *common.Request) ([]common.ResourceStatus, error) {
+	// Get the list of CRDs and check which legacy CRDs are present
+	crds := &unstructured.UnstructuredList{}
+	crds.SetKind("CustomResourceDefinition")
+	crds.SetAPIVersion("apiextensions.k8s.io/v1")
+	crdsPresent := make([]bool, len(sspOperands)) // All default to false
+	err := sspRequest.Client.List(sspRequest.Context, crds)
+	if err == nil {
+		for _, item := range crds.Items {
+			name := item.GetName()
+			for i, s := range kvsspCRDs {
+				if s == name {
+					crdsPresent[i] = true
+					break
+				}
+			}
+		}
+	}
+
 	allStatuses := make([]common.ResourceStatus, 0, len(sspOperands))
-	for _, operand := range sspOperands {
+	for i, operand := range sspOperands {
+		if crdsPresent[i] {
+			err := operand.PauseCRs(sspRequest)
```

Comment:
I think you can decouple the `PauseCRs` functions from the operands. They don't need to be related. The pausing code can be moved to this file or somewhere else, but it does not need to be part of the `Operand` interface.